hedera-node/src/main/java/com/hedera/services/stream/RecordStreamFileWriter.java

```diff
@@ -375,21 +401,50 @@ private void consume(final RecordStreamObject object) {
 						.build()
 		);
 
-		// In phase 2 we will need to add logic here to also
-		// save the sidecar records of the current object
+		final var sidecars = object.getSidecars();
+		if (isNotEmpty(sidecars)) {
+			for (final var sidecar : sidecars) {
+				switch (sidecar.getSidecarRecordsCase()) {
+					case STATE_CHANGES -> stateChangesSidecarBuilder.addSidecarRecords(sidecar);
+					case ACTIONS -> actionsSidecarBuilder.addSidecarRecords(sidecar);
+					case BYTECODE -> bytecodesSidecarBuilder.addSidecarRecords(sidecar);
+					case SIDECARRECORDS_NOT_SET -> LOG.warn("A sidecar record without an actual sidecar has been " +
+							"received");
+				}
```

Comment:
Maybe add a default case and log a warning?

Reply:
Agreed --- removed the `SIDECARRECORDS_NOT_SET` case and added a `default` that will catch all non-supported scenarios.